File: backend/entity/review.go
package entity

import "gorm.io/gorm"

// รีวิวของผู้ใช้ต่อเกมหนึ่ง ๆ
type Review struct {
	gorm.Model

	ReviewTitle string `json:"review_title" gorm:"type:varchar(255);not null"`
	ReviewText  string `json:"review_text" gorm:"type:text"`
	Rating      int    `json:"rating" gorm:"not null"` // 1–5

	// เปลี่ยนชื่อ composite unique index -> ux_reviews_user_game
	UserID uint  `json:"user_id" gorm:"not null;uniqueIndex:ux_reviews_user_game,priority:1"`
	User   *User `gorm:"foreignKey:UserID" json:"user"`

	GameID uint  `json:"game_id" gorm:"not null;uniqueIndex:ux_reviews_user_game,priority:2"`
	Game   *Game `gorm:"foreignKey:GameID" json:"game"`

	Likes []Review_Like `gorm:"foreignKey:ReviewID" json:"likes"`
}

func (Review) TableName() string { return "reviews" }

File: frontend/src/interfaces/Review.ts
// src/interfaces/Review.ts
import type { User } from "./User";
import type { Game } from "./Game";

export interface Review {
  ID: number;
  review_title: string;
  review_text?: string;
  rating: number;        // 0-10
  user_id: number;
  user?: User;
  game_id: number;
  game?: Game;
  likes?: ReviewLike[];
}

export interface ReviewLike {
  ID: number;
  review_id: number;
  user_id: number;
  game_id?: number;
}

export interface CreateReviewRequest {
  review_title: string;
  review_text?: string;
  rating: number;        // 0-10
  user_id: number;
  game_id: number;
}

export interface UpdateReviewRequest extends Partial<CreateReviewRequest> {}

export interface ToggleLikeRequest { user_id: number; }
